tests/objects/test_label_structure.py

```diff
@@ -640,6 +643,63 @@ def test_add_and_get_classification_instances_to_label_row(ontology):
     overlapping_classification_instance.set_for_frames(3)
 
 
+def test_add_and_get_classification_instances_to_audio_label_row(ontology):
+
+    label_row_metadata_dict = asdict(FAKE_LABEL_ROW_METADATA)
+    label_row_metadata_dict["frames_per_second"] = 1000
+    label_row_metadata_dict["data_type"] = "AUDIO"
+    label_row_metadata = LabelRowMetadata(**label_row_metadata_dict)
+
+    label_row = LabelRowV2(label_row_metadata, Mock(), ontology)
+    label_row.from_labels_dict(EMPTY_AUDIO_LABELS)
+
+    classification_instance_1 = ClassificationInstance(text_classification)
+    classification_instance_2 = ClassificationInstance(text_classification)
+    classification_instance_3 = ClassificationInstance(checklist_classification)
+
+    classification_instance_1.set_for_frames(Range(1, 2))
+    classification_instance_2.set_for_frames(Range(3, 4))
+    classification_instance_3.set_for_frames(Range(1, 4))
+
+    label_row.add_classification_instance(classification_instance_1)
+    label_row.add_classification_instance(classification_instance_2)
+    label_row.add_classification_instance(classification_instance_3)
+
+    classification_instances = label_row.get_classification_instances()
+    assert set(classification_instances) == {
+        classification_instance_1,
+        classification_instance_2,
+        classification_instance_3,
+    }
+
+    filtered_classification_instances = label_row.get_classification_instances(text_classification)
+    assert set(filtered_classification_instances) == {classification_instance_1, classification_instance_2}
+
+    overlapping_classification_instance = ClassificationInstance(text_classification)
+    overlapping_classification_instance.set_for_frames(1)
+
+    with pytest.raises(LabelRowError):
+        label_row.add_classification_instance(overlapping_classification_instance)
+
+    overlapping_classification_instance.remove_from_frames(1)
+    overlapping_classification_instance.set_for_frames(5)
+    label_row.add_classification_instance(overlapping_classification_instance)
+    with pytest.raises(LabelRowError):
```

Comment:
Wow nice. We write to the label row by reference rather than value. Somewhat of a footgun but great test!

Reply:
Consider either commenting this test or splitting it up a bit personally.
The continual mutated state is hard to follow. But test is good and operate at your own discretion.

Reply:
Agreed, but I copied another test and just edited it here.

I was also contemplating refactoring the tests to make it more readable, but fell on the side of lazy sadly.

But your comment has now tipped the scales. Will refactor!